src/internal/operators/takeLast.ts

```diff
@@ -46,36 +46,40 @@ export function takeLast<T>(count: number): MonoTypeOperatorFunction<T> {
   return count <= 0
     ? () => EMPTY
     : operate((source, subscriber) => {
-        // This buffer will hold the values we are going to emit
-        // when the source completes. Since we only want to take the
-        // last N values, we can't emit until we're sure we're not getting
-        // any more values.
-        let buffer: T[] = [];
-        source.subscribe(
-          createOperatorSubscriber(
-            subscriber,
-            (value) => {
-              // Add the most recent value onto the end of our buffer.
-              buffer.push(value);
-              // If our buffer is now larger than the number of values we
-              // want to take, we remove the oldest value from the buffer.
-              count < buffer.length && buffer.shift();
-            },
-            () => {
-              // The source completed, we now know what are last values
-              // are, emit them in the order they were received.
-              for (const value of buffer) {
-                subscriber.next(value);
-              }
-              subscriber.complete();
-            },
-            // Errors are passed through to the consumer
-            undefined,
-            () => {
-              // During finalization release the values in our buffer.
-              buffer = null!;
+      // This is a ring buffer that will hold our values
+      let ring = new Array(count)
```

Comment:
<img width="195" alt="any-array" src="https://user-images.githubusercontent.com/10235949/226887578-6bfdb506-9c3f-416e-ba12-bbf6059ef580.png">

I propose to explicitly specify the type parameter. This will ensure that only `T` is assigned to the array.

```suggestion
      let ring = new Array<T>(count)
```